src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -9,9 +10,34 @@ namespace GraphQL.Execution
     /// </summary>
     public static class ExecutionHelper
     {
+        private static readonly IDictionary<string, ArgumentValue> _emptyDirectiveArguments = new ReadOnlyDictionary<string, ArgumentValue>(new Dictionary<string, ArgumentValue>());
+
+        /// <summary>
+        /// Returns a dictionary of directives with their arguments values for a field.
+        /// Values will be retrieved from literals or variables as specified by the document.
+        /// </summary>
+        public static IDictionary<string, IDictionary<string, ArgumentValue>>? GetDirectives(GraphQLField field, Variables? variables, ISchema schema)
+        {
+            if (field.Directives == null || field.Directives.Count == 0)
+                return null;
```

Comment:
We could return an empty list here also.  It just saves a null check later at the cost of looking for a directive inside an empty list.  50-50 on this as well.  What you have is fine, as it matches `IResolveFieldContext.Arguments`, I believe.